pymc3/aesaraf.py

```diff
@@ -156,6 +156,10 @@ def change_rv_size(
             size = rv_node.op._infer_shape(size, dist_params)
         new_size = tuple(np.atleast_1d(new_size)) + tuple(size)
 
+    # Make sure the new size is int64 so that it doesn't unnecessarily pick
+    # up a `Cast` in some cases
+    new_size = at.as_tensor(new_size, ndim=1, dtype="int64")
```

Comment:
If the `new_size` is already a tensor variable, this line does not make sure it becomes `int64`:
```
at.as_tensor(at.constant([1,2], dtype="int32"), dtype="int64").dtype
# > "int32"
```
However it still makes #4652 less likely to occur.

```suggestion
    # Make sure the new size is a tensor. This helps to not unnecessarily pick
    # up a `Cast` in some cases
    new_size = at.as_tensor(new_size, ndim=1, dtype="int64")
```